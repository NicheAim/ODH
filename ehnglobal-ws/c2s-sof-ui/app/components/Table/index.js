/**
 *
 * Table
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const Table = styled.div`
  margin: 0 10px;
  background-color: white;
`;


Table.propTypes = {
  children: PropTypes.node,
};

export default Table;
