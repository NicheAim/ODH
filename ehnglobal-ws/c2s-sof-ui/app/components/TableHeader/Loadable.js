/**
 *
 * Asynchronously loads the component for TableHeader
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
