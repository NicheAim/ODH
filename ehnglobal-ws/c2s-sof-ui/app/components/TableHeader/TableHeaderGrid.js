import { Grid } from 'styled-css-grid';

const TableHeaderGrid = Grid.extend`
  border-bottom: 2px outset #000;

  &:first-child {
    background-color: #fff;
    font-size: 20px;
  }

  &:nth-child(odd) {
    background-color: #fff;
  }
`;

TableHeaderGrid.propTypes = {};

export default TableHeaderGrid;
