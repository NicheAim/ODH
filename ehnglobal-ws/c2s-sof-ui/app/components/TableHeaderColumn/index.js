/**
 *
 * TableHeaderColumn
 *
 */


import PropTypes from 'prop-types';
import styled from 'styled-components';

const TableHeaderColumn = styled.div`
  color: #000;
  font-family: "Arial Bold", "Arial", sans-serif;
  font-style: normal;
  font-size: 12px;
  font-weight: 700;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-all;
  padding: 2px;
`;

TableHeaderColumn.propTypes = {
  children: PropTypes.node,
};

export default TableHeaderColumn;
