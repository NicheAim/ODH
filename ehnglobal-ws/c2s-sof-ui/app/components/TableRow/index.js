/**
 *
 * TableRow
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import pickBy from 'lodash/pickBy';

import TableRowGrid from './TableRowGrid';

function TableRow({ children, columns, onClick, onKeyPress, role, tabIndex }) {
  const additionalProps = pickBy({ onClick, onKeyPress, role, tabIndex });
  return (
    <TableRowGrid
      gap="5px"
      columns={columns || `repeat(${React.Children.count(children)}, 1fr)`}
      {...additionalProps}
    >
      {children}
    </TableRowGrid>
  );
}

TableRow.propTypes = {
  children: PropTypes.node,
  columns: PropTypes.string,
  onClick: PropTypes.func,
  onKeyPress: PropTypes.func,
  role: PropTypes.string,
  tabIndex: PropTypes.string,
};

export default TableRow;
