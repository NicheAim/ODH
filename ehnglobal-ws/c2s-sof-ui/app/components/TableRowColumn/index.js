/**
 *
 * TableRowColumn
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const TableRowColumn = styled.div`
  color: #000;
  font-family: "Arial", sans-serif;
  font-style: normal;
  font-size: 12px;
  font-weight: 400;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-all;
  padding: 5px;
`;

TableRowColumn.propTypes = {
  children: PropTypes.node,
};

export default TableRowColumn;
