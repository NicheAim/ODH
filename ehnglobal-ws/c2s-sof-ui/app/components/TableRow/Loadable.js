/**
 *
 * Asynchronously loads the component for TableRow
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
