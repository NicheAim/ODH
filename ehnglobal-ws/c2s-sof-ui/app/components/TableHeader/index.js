/**
 *
 * TableHeader
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import StickyDiv from 'components/StickyDiv';
import TableHeaderGrid from './TableHeaderGrid';

function TableHeader(props) {
  const { sticky, children, columns, relativeTop } = props;

  function renderTableHeader() {
    return (
      <TableHeaderGrid gap="5px" columns={columns || `repeat(${React.Children.count(children)}, 1fr)`}>
        {children}
      </TableHeaderGrid>
    );
  }

  let renderContent = renderTableHeader();
  if (sticky) {
    renderContent = (
      <StickyDiv top={`${relativeTop}px`}>
        {renderTableHeader()}
      </StickyDiv>
    );
  }

  return renderContent;
}

TableHeader.propTypes = {
  sticky: PropTypes.bool,
  relativeTop: PropTypes.number.isRequired,
  children: PropTypes.node,
  columns: PropTypes.string,
};

TableHeader.defaultProps = {
  sticky: true,
  relativeTop: 0,
};

export default TableHeader;
