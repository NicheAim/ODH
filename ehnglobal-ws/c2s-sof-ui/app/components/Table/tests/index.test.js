import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import Table from '../index';
import TableHeaderColumn from '../../TableHeaderColumn';
import TableHeader from '../../TableHeader';
import TableRow from '../../TableRow';
import TableRowColumn from '../../TableRowColumn';

configure({ adapter: new Adapter() });

describe('<Table />', () => {
  it('should match snapshot', () => {
    // Act
    const renderedComponent = shallow(
      (<Table>
        <TableHeader>
          <TableHeaderColumn>Col</TableHeaderColumn>
        </TableHeader>
        <TableRow>
          <TableRowColumn>Row</TableRowColumn>
        </TableRow>
      </Table>),
    );

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
})
;
