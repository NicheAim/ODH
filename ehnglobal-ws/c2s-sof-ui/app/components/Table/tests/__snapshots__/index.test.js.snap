// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table__Table>
    <TableHeader
      relativeTop={0}
      sticky={true}
    >
      <TableHeaderColumn__TableHeaderColumn>
        Col
      </TableHeaderColumn__TableHeaderColumn>
    </TableHeader>
    <TableRow>
      <TableRowColumn__TableRowColumn>
        Row
      </TableRowColumn__TableRowColumn>
    </TableRow>
  </Table__Table>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TableHeader
          relativeTop={0}
          sticky={true}
        >
          <TableHeaderColumn__TableHeaderColumn>
            Col
          </TableHeaderColumn__TableHeaderColumn>
        </TableHeader>,
        <TableRow>
          <TableRowColumn__TableRowColumn>
            Row
          </TableRowColumn__TableRowColumn>
        </TableRow>,
      ],
      "className": "Table__Table-fLBySN dazbIp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <TableHeaderColumn__TableHeaderColumn>
            Col
          </TableHeaderColumn__TableHeaderColumn>,
          "relativeTop": 0,
          "sticky": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Col",
          },
          "ref": null,
          "rendered": "Col",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <TableRowColumn__TableRowColumn>
            Row
          </TableRowColumn__TableRowColumn>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Row",
          },
          "ref": null,
          "rendered": "Row",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TableHeader
            relativeTop={0}
            sticky={true}
          >
            <TableHeaderColumn__TableHeaderColumn>
              Col
            </TableHeaderColumn__TableHeaderColumn>
          </TableHeader>,
          <TableRow>
            <TableRowColumn__TableRowColumn>
              Row
            </TableRowColumn__TableRowColumn>
          </TableRow>,
        ],
        "className": "Table__Table-fLBySN dazbIp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <TableHeaderColumn__TableHeaderColumn>
              Col
            </TableHeaderColumn__TableHeaderColumn>,
            "relativeTop": 0,
            "sticky": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Col",
            },
            "ref": null,
            "rendered": "Col",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <TableRowColumn__TableRowColumn>
              Row
            </TableRowColumn__TableRowColumn>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Row",
            },
            "ref": null,
            "rendered": "Row",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
